cfme/tests/containers/test_table_views.py

```diff
@@ -15,75 +12,60 @@
-    original_default_views, tested_default_views = OrderedDict(), OrderedDict()
-    for collection_name, ui_name in objects_mapping.items():
-        original_default_views[collection_name] = (
-            appliance.user.my_settings.default_views.get_default_view(ui_name))
-        tested_default_views[collection_name] = choice(VIEWS)
-    appliance.user.my_settings.default_views.set_default_view(objects_mapping.values(),
-                                                              tested_default_views.values())
-    yield tested_default_views
-    # setting back the default views to the original state:
-    appliance.user.my_settings.default_views.set_default_view(objects_mapping.values(),
-                                                              original_default_views.values())
+objects_mapping = {
+    'Containers Providers': ContainersProvider,
+    'Nodes': 'container_nodes',
+    'Pods': 'container_pods',
+    'Services': 'container_services',
+    'Routes': 'container_routes',
+    'Containers': 'containers',
+    'Projects': 'container_projects',
+    'Replicators': 'container_replicators',
+    'Container Images': 'container_images',
+    'Image Registries': 'container_image_registries',
+}
 
 
-def test_default_views(appliance, random_default_views):
+@pytest.mark.parametrize('group_name', list(objects_mapping.keys()),
+                         ids=["_".join(gp.split()) for gp in objects_mapping], scope="module")
+@pytest.mark.parametrize('new_default_view', ['List View', 'Tile View', 'Grid View'],
+                         ids=['List_View', 'Tile_View', 'Grid_View'])
+def test_default_views(appliance, group_name, new_default_view):
     """
     Polarion:
         assignee: juwatts
         caseimportance: high
         casecomponent: Containers
         initialEstimate: 1/6h
     """
-    for collection_name in objects_mapping.keys():
-        obj = (ContainersProvider if collection_name is ContainersProvider
-               else getattr(appliance.collections, collection_name))
-        view = navigate_to(obj, 'All', use_resetter=False)
-        assert (random_default_views[collection_name].lower() ==
-                view.toolbar.view_selector.selected.lower()), (
-            "Failed to setup default view \"{}\" for {}".format(
-                view, objects_mapping[collection_name])
-        )
+    collection_name = objects_mapping[group_name]
+    default_views = appliance.user.my_settings.default_views
+    orig_default = default_views.get_default_view(group_name)
+    default_views.set_default_view(group_name, new_default_view)
+    obj = (ContainersProvider if collection_name is ContainersProvider
+           else getattr(appliance.collections, collection_name))
+    view = navigate_to(obj, 'All', use_resetter=False)
+    assert view.toolbar.view_selector.selected == new_default_view
+    default_views.set_default_view(group_name, orig_default)
```

Comment:
Think this should be called with request.addfinalizer() in case test validation/steps fail.